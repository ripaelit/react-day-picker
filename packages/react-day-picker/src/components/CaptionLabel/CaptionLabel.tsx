import React from 'react';

import { useDayPicker } from '../../hooks';
import { UIElement as UI } from '../../types';

/** The props for the [[CaptionLabel]] component. */
export interface CaptionLabelProps {
  /** The month where the caption is displayed. */
  displayMonth: Date;
}

export function CaptionLabel(props: CaptionLabelProps): JSX.Element {
  const {
    locale,
    classNames,
    styles,
    formatters: { formatCaption }
  } = useDayPicker();
  return (
    <div
      key="caption"
      className={classNames[UI.CaptionLabel]}
      style={styles[UI.CaptionLabel]}
      aria-live="polite"
    >
      {formatCaption(props.displayMonth, { locale })}
    </div>
  );
}
