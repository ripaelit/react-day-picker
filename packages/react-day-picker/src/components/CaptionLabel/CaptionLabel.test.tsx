import { RenderResult } from '@testing-library/react';
import React from 'react';

import { customRender } from 'test';
import tk from 'timekeeper';

import { CaptionLabel } from './CaptionLabel';
import { CaptionLabelProps } from './CaptionLabelProps';

const FrozenDate = new Date(2020, 5);

beforeEach(() => tk.freeze(FrozenDate));
afterEach(() => tk.reset());

const setup = (props?: CaptionLabelProps): RenderResult => {
  return customRender(<CaptionLabel displayMonth={new Date()} {...props} />);
};
test('should render correctly', () => {
  const { container } = setup();
  expect(container.firstChild).toMatchSnapshot();
});
