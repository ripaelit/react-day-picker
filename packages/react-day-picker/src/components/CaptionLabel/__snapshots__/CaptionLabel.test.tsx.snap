// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  aria-live="polite"
  class="rdp-caption_label"
>
  June 2020
</div>
`;
